stdlib/_ast.pyi

```diff
@@ -346,19 +593,24 @@ class Constant(expr):
     # Aliases for value, for backwards compatibility
     s: Any
     n: int | float | complex
+    def __init__(self, value: Any, kind: str | None = None, **kwargs: typing_extensions.Unpack[_Attributes]) -> None: ...
 
 class NamedExpr(expr):
     if sys.version_info >= (3, 10):
         __match_args__ = ("target", "value")
     target: Name
     value: expr
+    def __init__(self, target: Name, value: expr, **kwargs: typing_extensions.Unpack[_Attributes]) -> None: ...
 
 class Attribute(expr):
     if sys.version_info >= (3, 10):
         __match_args__ = ("value", "attr", "ctx")
     value: expr
     attr: _Identifier
     ctx: expr_context
+    def __init__(
+        self, value: expr, attr: _Identifier, ctx: expr_context, **kwargs: typing_extensions.Unpack[_Attributes]
```

Comment:
Would it make sense to make the `ctx` argument optional in constructor methods? Code like this in flake8-pyi is being flagged currently by mypy_primer, and it seems reasonable to allow constructing AST nodes on the fly without passing in a `ctx` argument: https://github.com/PyCQA/flake8-pyi/blob/f60d7e3cdc833dd358b493d2ca80fd187cc38cfb/pyi.py#L1463

```suggestion
        self, value: expr, attr: _Identifier, ctx: expr_context = ..., **kwargs: typing_extensions.Unpack[_Attributes]
```

Reply:
I tend to agree with this suggestion - but I'm looking at the recent commits on Python 3.13 which are now issuing `DeprecationWarning`; `ctx` is not spared from this:

```pycon
>>> ast.Attribute()
<python-input-5>:1: DeprecationWarning: Attribute.__init__ missing 1 required positional argument: 'ctx'. This will become an error in Python 3.15.
  ast.Attribute()
<python-input-5>:1: DeprecationWarning: Attribute.__init__ missing 1 required positional argument: 'attr'. This will become an error in Python 3.15.
  ast.Attribute()
<python-input-5>:1: DeprecationWarning: Attribute.__init__ missing 1 required positional argument: 'value'. This will become an error in Python 3.15.
  ast.Attribute()
<ast.Attribute object at 0x7216fd006f10>
```

Not sure what the best approach is here - whether to be permissive with a sudden hit in Python 3.13, or to warn about this now?

Reply:
Hmm. @JelleZijlstra I think the deprecation warning could be quite disruptive for `ctx` parameters specifically. Do you think we could give the parameter a default value over at CPython?

Reply:
Yes, probably.

Reply:
https://github.com/python/cpython/pull/118854

Reply:
Addressed in `https://github.com/python/typeshed/pull/11880/commits/bb880f2031db86161251e580e0a37429395bde3c`